Kernel/Memory/MemoryManager.cpp

```diff
@@ -416,7 +416,7 @@ UNMAP_AFTER_INIT void MemoryManager::initialize_physical_pages()
 {
     m_global_data.with([&](auto& global_data) {
         // We assume that the physical page range is contiguous and doesn't contain huge gaps!
-        PhysicalAddress highest_physical_address;
+        PhysicalAddress highest_physical_address = PhysicalAddress { 0x100000000LL };
```

Comment:
I've seen this kind of behavior on other bare metal machines, but I am not sure if we want to start with `highest_physical_address` being at 4GB... could you provide the memory map you have on that machine?

Reply:
Problem is with LAPIC that is mapped at almost 4 GiB. And it's not in the memory map.
My memory map:
0-4KiB: core boot tables anchor (RAM)
4KiB-640KiB: RAM
640KiB-1MiB: reserved
1 MiB-1867MiB:RAM including ACPI tables
1867MiB-2048MiB: runtime services, core boot tables and so on
4GiB-10GiB: RAM

2GiB-4GiB is used for various devices, not listed in memory map. 2GiB is used for framebuffer.

4GiB guarantees that all kind of LAPIC and devices are mapped